storage/client/client.go

```diff
@@ -1257,3 +1257,51 @@ func (c *StorageClient) TxVolumes(ctx context.Context, layer apiTypes.Layer, p a
 
 	return &ts, nil
 }
+
+// DailyActiveAccounts returns a list of daily active accounts.
+func (c *StorageClient) DailyActiveAccounts(ctx context.Context, layer apiTypes.Layer, p apiTypes.GetLayerStatsActiveAccountsParams) (*DailyActiveAccountsList, error) {
+	var query string
+	switch {
+	case p.WindowStepSeconds != nil && *p.WindowStepSeconds == 300:
+		query = QueryFactoryFromCtx(ctx).FineDailyActiveAccountsQuery()
+	default:
+		query = QueryFactoryFromCtx(ctx).DailyActiveAccountsQuery()
+	}
+
+	rows, err := c.db.Query(
+		ctx,
+		query,
+		layer,
+		p.Limit,
+		p.Offset,
+	)
+	if err != nil {
+		return nil, wrapError(err)
+	}
+	defer rows.Close()
+
+	ts := DailyActiveAccountsList{
+		WindowSizeSeconds: 86400, // Day.
+		Windows:           []apiTypes.ActiveAccounts{},
+	}
+	for rows.Next() {
+		var d struct {
```

Comment:
Optional: `d` could be directly `apiTypes.ActiveAccounts`. You'd just need to modify the query to do the UTC conversion in postgres:
`SELECT window_end AT TIME ZONE 'UTC' AT TIME ZONE 'UTC'`

That said, the syntax is pretty stupid. The first `AT TIME ZONE` converts to a timestamp without timezone, the second one re-adds the UTC timezone. See [docs](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-ZONECONVERT). Your call whether we should go for it.

Reply:
Thanks for the suggestion. I have mixed feeling about this, on one hand It would be nice to use  `apiTypes.ActiveAccounts` directly, but I feel that the `"AT TIME ZONE 'UTC' AT TIME ZONE 'UTC'"` merits a doc comment explaining it. But then there are 4 queries where this should be used (2 for active accounts and 2 for tx volume), so maybe there can be a helper function in the `queries.go` to append this.

But there is also a different way to avoid the `d` helper struct, I'm thinking the nicest solution for all this is to just do:

```go
var t apiTypes.ActiveAccounts
if err := rows.Scan(
	&t.WindowEnd,
	&t.ActiveAccounts,
); err != nil {
	return nil, wrapError(err)
}
t.WindowEnd = t.WindowEnd.UTC() // Ensure timestamp is in UTC.

ts.Windows = append(ts.Windows, t)
```